teps/0122-complete-build-instructions-and-parameters.md

```diff
@@ -0,0 +1,241 @@
+- **Complete list of system architecture** like host architecture (amd64 vs arm64), SELinux, container runtimes, and kubernetes version are out-of-scope.
+- **Tekton system versioning** since there are a couple of separate efforts trying to tackle this (eg, [TEP-0041](https://github.com/tektoncd/community/blob/main/teps/0041-tekton-component-versioning.md), [tektoncd/pipeline#5303](https://github.com/tektoncd/pipeline/issues/5303)).
+
+## Proposal
+For the completeness of build instructions, we need to be able to extract enough information from the provenance to recreate the task run or pipeline run under the same set of configuration as was used when creating the build. The list below shows which fields are required in the provenance for re-creating the task run.
+- [Task Specification](#task-specification)
+- [Step Specification](#step-specification)
+- [TaskRef](#taskref)
+- [TaskRun Specification](#taskrun-specification)
+- [Configuration feature flags](#configuration-feature-flags)
+- [Materials](#materials)
+
+This will allow us to create a provenance for a task run as [shown](#provenance-for-executed-taskrun).
+
+### Task Specification
+The table below provides the name and description of the API fields that the Task requires. In addition, it also lists if that information is required, not required, or provided by the provenance. If it is required, it also shows where the information should be provided. Looking at the table below and required [step](#step-specification) information, there are only a handful of fields that are not required for reproducibility. Instead of cherry-picking most of the fields, including the entire taskSpec in the buildConfig section of the provenance is likely the best way to go forward. Additionally, if we update the spec in the future, then chains will not have to be updated correspondingly. The work described in [issue](https://github.com/tektoncd/chains/issues/597) is meant to handle this.
+
+
+| | **Metadata**      | | |
+| ------ | ------------- | ----- | ----- |
+| **Field Name**    | **Description** | **Not Required / Required / Provided by provenance** | **Insert in provenance** |
+| name          | Name of the task | Not Required (if the spec is already in the provenance) | |
```

Comment:
NIT: We provide a variable to access the task name within the task `context.task.name` which means that the task name could be used in the build steps and should be part of the provenance

Reply:
good point. This strengthens my case even more that there are barely any fields that are not required for reproducibility and so we should just add the entire task spec. I will update it.